programs/programs/testing_fixtures/custom_calculator.py

```
@@ -0,0 +1,225 @@
+        `Program.objects.new_program` writes a `Translation` row per translated field per
+        language, so building it per test method dominates the runtime of a small suite.
+        Neither row is mutated by a test, so one per class is enough.
+        """
+        super().setUpTestData()
+        cls.white_label = make_white_label(cls.white_label_code, cls.state_code)
+        if cls.needs_program_row:
+            cls.program = make_program(cls.white_label_code, cls.program_code, cls.fpl_year, cls.state_code)
+        else:
+            cls.program = Mock()
+
+    def make_calculator(self, screen: Screen, data: dict[str, Eligibility] = None, missing=()):
```

Ruff (0.16.1)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)